pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -334,6 +340,74 @@ public void indexRow(GenericRow row)
     }
 
     _docIdCounter++;
+    _durationNS += System.nanoTime() - startNS;
+  }
+
+  @Override
+  public void indexColumn(String columnName, @Nullable int[] sortedDocIds, IndexSegment segment)
+          throws IOException {
+    long startNS = System.nanoTime();
+
+    // TODO(ERICH): Get a measure of the ratio of columns to indexes (how many indexes per column are there)
+    // Iterate over each value in the column
+    try (PinotSegmentColumnReader colReader = new PinotSegmentColumnReader(segment, columnName)) {
+      int numDocs = segment.getSegmentMetadata().getTotalDocs();
+      Map<IndexType<?, ?, ?>, IndexCreator> creatorsByIndex = _creatorsByColAndIndex.get(columnName);
+      NullValueVectorCreator nullVec = _nullValueVectorCreatorMap.get(columnName);
+      if (sortedDocIds != null) {
+        for (int docId : sortedDocIds) {
+          // TODO(Erich): should I avoid a function call in the loop like this?
+          indexColumnValue(colReader, creatorsByIndex, columnName, docId, nullVec);
```

Comment:
Function call in the loop is fine. Let's extract the per column logic (e.g. map lookups) outside of this method to reduce the cost